xtask/bench/src/lib.rs

```
@@ -1,90 +1,76 @@
-use ansi_rgb::{red, Foreground};
-use itertools::Itertools;
-use rslint_errors::Diagnostic;
+mod features;
+mod utils;
+
 use std::fmt::{Display, Formatter};
-use std::ops::Add;
+use std::str::FromStr;
 use std::time::Duration;
-use std::{path::PathBuf, str::FromStr};
 
-fn err_to_string<E: std::fmt::Debug>(e: E) -> String {
-    format!("{:?}", e)
+pub use crate::features::formatter::benchmark_format_lib;
+use crate::features::formatter::BenchmarkFormatterResult;
+pub use crate::features::parser::benchmark_parse_lib;
+use crate::features::parser::BenchmarkParseResult;
+pub use utils::get_code;
+
+/// What feature to benchmark
+pub enum FeatureToBenchmark {
+    /// benchmark of the parser
+    Parser,
+    /// benchmark of the formatter
+    Formatter,
+}
+
+impl FromStr for FeatureToBenchmark {
+    type Err = ();
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "parser" => Ok(Self::Parser),
+            "formatter" => Ok(Self::Formatter),
+            _ => Ok(Self::Parser),
```

I feel this should be an error, if you pass `--feature formater` having it silently default to the parser can be confusing